File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/PrefChangeEvent.java
package work.socialhub.kslack.entity.event;

/**
 * The pref_change event is sent to all connections for a user when a user preference is changed.
 * Clients should update to reflect new changes immediately.
 * <p>
 * https://api.slack.com/events/pref_change
 */
public class PrefChangeEvent implements Event {

    public static final String TYPE_NAME = "pref_change";

    private final String type = TYPE_NAME;
    private String name;
    private String value;

    public String getType() {
        return this.type;
    }

    public String getName() {
        return this.name;
    }

    public String getValue() {
        return this.value;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setValue(String value) {
        this.value = value;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/PrefChangeEvent.kt
package work.socialhub.kslack.entity.event

/**
 * The pref_change event is sent to all connections for a user when a user preference is changed.
 * Clients should update to reflect new changes immediately.
 *
 *
 * https://api.slack.com/events/pref_change
 */
class PrefChangeEvent : Event {
    override val type: String = TYPE_NAME
    var name: String? = null
    var value: String? = null

    companion object {
        const val TYPE_NAME: String = "pref_change"
    }
}